scripts/dirty-check.js

```diff
@@ -0,0 +1,70 @@
+// @ts-check
+
+/**
+ * This script is intended to be run in a GitHub Actions workflow to check if
+ * test execution changed any files under version control. If so, the script
+ * will exit with a non-zero exit code and print an "error" to the GitHub
+ * Actions log for each changed file.
+ *
+ * This is not a shell script because portability.
+ *
+ * @example
+ *
+ * ```yaml
+ * on: [push, pull_request]
+ * jobs:
+ *   test:
+ *     runs-on: ubuntu-latest
+ *   steps:
+ *     - run: your stuff
+ *     - run: node ./scripts/dirty-check.js
+ * ```
+ *
+ * @packageDocumentation
+ * @see
+ * {@link https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions}
+ */
+
+const { promisify } = require('node:util')
+const execFile = promisify(require('node:child_process').execFile)
+
+/**
+ * Asks `git` if any files changed and exits with a non-zero exit code if so.
+ *
+ * @returns {Promise<void>}
+ * @todo We could determine what lines changed in what file so the file would be
+ *   annotated with the error, but it requires some extra parsing of `git diff`
+ *   output.
+ */
+async function main() {
+  const { stdout } = await execFile('git', ['diff', '-b', '--name-only'], {
```

Comment:
```suggestion
  const { stdout } = await execFile('git', ['diff', '-w', '-b', '--name-only'], {
```

if the remaining issue is simply whitespace-related, ignoring whitespace-only changes could be a temporary workaround?

i guess a proper solution involves updating `.gitattributes`

Reply:
This is one of the problems, yes. 

I was having a tough time reproducing this locally, since it seems to be linux-only, and Docker Desktop for M2 is broken, so I can't run `act` to quickly iterate.